dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/OperationsParamsValidation.java

```diff
@@ -0,0 +1,222 @@
+      throws ForbiddenException {
+
+    Set<OrganisationUnit> accessibleOrgUnits = new HashSet<>();
+
+    for (OrganisationUnit orgUnit : orgUnits) {
+      Set<OrganisationUnit> accessibleOrgUnitsFound =
+          switch (orgUnitMode) {
+            case DESCENDANTS -> getAccessibleDescendants(
+                user, program, orgUnitDescendants.apply(orgUnit.getUid()));
+            case CHILDREN -> getAccessibleDescendants(
+                user,
+                program,
+                Stream.concat(Stream.of(orgUnit), orgUnit.getChildren().stream()).toList());
+            case SELECTED -> getSelectedOrgUnits(user, program, orgUnit, trackerAccessManager);
+            default -> Collections.emptySet();
+          };
+
+      if (accessibleOrgUnitsFound.isEmpty()) {
+        throw new ForbiddenException("User does not have access to orgUnit: " + orgUnit.getUid());
+      }
+
+      accessibleOrgUnits.addAll(accessibleOrgUnitsFound);
+    }
+
+    if (orgUnitMode == CAPTURE) {
+      return new HashSet<>(user.getOrganisationUnits());
+    } else if (orgUnitMode == ACCESSIBLE) {
+      return getAccessibleOrgUnits(user, program);
+    }
+
+    return accessibleOrgUnits;
+  }
+
+  private static Set<OrganisationUnit> getSelectedOrgUnits(
+      User user,
+      Program program,
+      OrganisationUnit orgUnit,
+      TrackerAccessManager trackerAccessManager) {
+    return trackerAccessManager.canAccess(user, program, orgUnit)
+        ? Set.of(orgUnit)
+        : Collections.emptySet();
+  }
+
+  private static Set<OrganisationUnit> getAccessibleOrgUnits(User user, Program program) {
+    return isProgramAccessRestricted(program)
+        ? new HashSet<>(user.getOrganisationUnits())
+        : new HashSet<>(user.getTeiSearchOrganisationUnitsWithFallback());
+  }
+
+  /**
+   * Returns the org units whose path is contained in the user search or capture scope org unit. If
+   * there's a match, it means the user org unit is at the same level or above the supplied org
+   * unit.
+   *
+   * @param user the user to check the access of
+   * @param program the program the user wants to access to
+   * @param orgUnits the org units to check if the user has access to
+   * @return a list with the org units the user has access to
+   */
+  private static Set<OrganisationUnit> getAccessibleDescendants(
+      User user, Program program, List<OrganisationUnit> orgUnits) {
+    if (orgUnits.isEmpty()) {
+      return Collections.emptySet();
+    }
+
+    if (isProgramAccessRestricted(program)) {
+      return orgUnits.stream()
+          .filter(
+              availableOrgUnit ->
+                  user.getOrganisationUnits().stream()
+                      .anyMatch(
+                          captureScopeOrgUnit ->
+                              availableOrgUnit.getPath().contains(captureScopeOrgUnit.getPath())))
+          .collect(Collectors.toSet());
+    } else {
+      return orgUnits.stream()
```

Comment:
nit: since we return in the else I would remove the else and return instead (intellij will likely show you that cleanup option)